src/main/user-impl/java/com/mysql/cj/jdbc/MysqlSQLXML.java

```diff
@@ -212,6 +218,16 @@ public synchronized Reader getCharacterStream() throws SQLException {
             try {
                 DocumentBuilderFactory builderFactory = DocumentBuilderFactory.newInstance();
                 builderFactory.setNamespaceAware(true);
+
+                if (!this.allowXmlUnsafeExternalEntity) {
```

Comment:
Can we have few unit tests to make sure it fixes the issue?

Reply:
Yeap for sure!